app/controllers/admin/problems_controller.rb

```diff
@@ -1,4 +1,57 @@
 # frozen_string_literal: true
 
 class Admin::ProblemsController < Admin::ApplicationController
+  before_action :set_session
+  before_action :set_mission
+  before_action :set_problem, only: %i[show]
+
+  def new
+    @problem = @mission.problems.new
+  end
+
+  def create
+    @problem = Problem.create(
+      mission: @mission,
+      name: params[:problem][:name],
+      detail: params[:problem][:detail],
+    )
+
+    test_command_list =  @mission.test_commands
+    params['pirced-location-id'].each_with_index do |id, index|
+      test_name = params[:tests][index]
+
+      test = Test.create(
+        mission: @mission,
+        test_name: test_name,
+        test_command: test_command_list[test_name],
+        pierced_location_id: id
+      )
+
+      ProblemTest.create(
+        problem_id: @problem.id,
+        test_id: test.id,
+        pierced_level: params[:labels][index]
+      )
+    end
+
+    redirect_to admin_session_mission_problems_path(params[:session_id], params[:mission_id], id: @problem.to_param)
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Metrics/LineLength: Line is too long. [116/100]